tests/sentry/workflow_engine/test_base.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from uuid import uuid4
+
+from sentry.eventstore.models import Event, GroupEvent
+from sentry.models.group import Group
+from sentry.snuba.models import SnubaQuery
+from sentry.testutils.cases import TestCase
+from sentry.testutils.factories import EventType
+from sentry.workflow_engine.models import DataConditionGroup, Detector, DetectorWorkflow, Workflow
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.types import DetectorType
+from tests.sentry.issues.test_utils import OccurrenceTestMixin
+
+
+class BaseWorkflowTest(TestCase, OccurrenceTestMixin):
+    def create_snuba_query(self, **kwargs):
+        return SnubaQuery.objects.create(
+            type=SnubaQuery.Type.ERROR.value,
+            dataset="events",
+            aggregate="count()",
+            time_window=60,
+            resolution=60,
+            **kwargs,
+        )
+
+    def create_event(
+        self,
+        project_id: int,
+        timestamp: datetime,
+        fingerprint: str,
+        environment=None,
+        tags: list[list[str]] | None = None,
+    ) -> Event:
+        data = {
+            "timestamp": timestamp.isoformat(),
+            "environment": environment,
+            "fingerprint": [fingerprint],
+            "level": "error",
+            "user": {"id": uuid4().hex},
+            "exception": {
+                "values": [
+                    {
+                        "type": "IntegrationError",
+                        "value": "Identity not found.",
+                    }
+                ]
+            },
+        }
+        if tags:
+            data["tags"] = tags
+
+        return self.store_event(
+            data=data,
+            project_id=project_id,
+            assert_no_errors=False,
+            default_event_type=EventType.ERROR,
+        )
+
+    def create_detector_and_workflow(
+        self,
+        name_prefix="test",
```

Comment:
Can this be inferred from the `detector_type`? I see in `TestProcessWorkflows` it's passing `"error"` as the `name_prefix` when it passes in the errors detector type, and for the issue occurrence one it doesn't pass either. Is there a time when we'd be passing something else?

Reply:
ah, yes, i'll remove. (i was using it for debugging and failed to clean it up, ty!)

Reply:
i just did this as a default name in tests, so i can identify them uniquely / easily, but wanted a default so i only have to add when i have multiple detectors.

if we based it on `detector_type` and had two detectors of the same time in a test, it might be a bit confusing.